src/v/raft/consensus.cc

```diff
@@ -741,7 +741,9 @@ consensus::linearizable_barrier(model::timeout_clock::time_point deadline) {
 
     // wait for responses in background
     ssx::spawn_with_gate(_bg, [futures = std::move(send_futures)]() mutable {
-        return ss::when_all_succeed(futures.begin(), futures.end());
```

Comment:
1. I don't think this changes the lifetimes. AFAICT `when_all_succeed` moves all of the futures out of the provided range into its own `vector<future>`. Can you say more about how you think the UAF occurs?
2. Unrelated, but I noticed: if `_bg` is closed and `send_futures` contains a failed future, then Seastar is going to complain about abandoned exceptional futures.

Reply:
> AFAICT when_all_succeed moves all of the futures out of the provided range into its own vector<future>

Ah good spot, I didn't bother looking into when_all_succeed, thoughts its iterating over the container.

>  if _bg is closed and send_futures contains a failed future, then Seastar is going to complain about abandoned exceptional futures.

🦅  👁️